provider/overlays.go

```diff
@@ -188,4 +188,11 @@ func init() {
 	resourceOverlays[callbackFunctionTok] = callbackFunction
 
 	typeOverlays[codePathOptionsTok] = codePathOptions
+
+	typeOverlays["aws:index/aRN:ARN"] = schema.ComplexTypeSpec{
```

Comment:
This is an attempt to register an overlay-only type with the schema, hoping that Node SDK would be able to recognize references and other SDK would ignore it.